bot/exts/backend/branding/_cog.py

```diff
@@ -1,566 +1,654 @@
+    """
+
+    # RedisCache[
+    #     "daemon_active": If True, daemon auto-starts; controlled via commands (bool)
+    #     "event_path": Path from root in the branding repo (str)
+    #     "event_description": Markdown description (str)
+    #     "event_duration": Human-readable date range or 'Fallback' (str)
+    #     "banner_hash": Hash of the last applied banner (str)
+    #     "icons_hash": Compound hash of icons in rotation (str)
+    #     "last_rotation_timestamp": POSIX timestamp (float)
+    # ]
+    cache_information = RedisCache()
+
+    # Cache holding icons in current rotation ~ the keys are download URLs (str) and the values are integers
+    # corresponding to the amount of times each icon has been used in the current rotation
+    cache_icons = RedisCache()
+
+    # Cache holding all available event names & their durations; this is cached by the daemon and read by
+    # the calendar command with the intention of preventing API spam; doesn't contain the fallback event
+    cache_events = RedisCache()
 
     def __init__(self, bot: Bot) -> None:
+        """Instantiate repository abstraction & allow daemon to start."""
+        self.bot = bot
+        self.repository = BrandingRepository(bot)
+
+        self.bot.loop.create_task(self.maybe_start_daemon())  # Start depending on cache
+
+    # region: Internal utility
+
+    @mock_in_debug(return_value=True)
+    async def apply_asset(self, asset_type: AssetType, download_url: str) -> bool:
         """
-        Assign safe default values on init.
+        Download asset from `download_url` and apply it to PyDis as `asset_type`.
 
-        At this point, we don't have information about currently available branding.
-        Most of these attributes will be overwritten once the daemon connects, or once
-        the `refresh` command is used.
+        This function is mocked in the development environment in order to prevent API spam during testing.
+        Decorator should be temporarily removed in order to test internal methodology.
+
+        Returns a boolean indicating whether the application was successful.
         """
-        self.bot = bot
-        self.current_season = _seasons.get_current_season()
+        log.info(f"Applying {asset_type.value} asset to the guild")
 
-        self.banner = None
+        try:
+            file = await self.repository.fetch_file(download_url)
+        except Exception as fetch_exc:
+            log.error(f"Failed to fetch '{asset_type.value}' asset: {fetch_exc}")
+            return False
 
-        self.available_icons = []
-        self.remaining_icons = []
+        await self.bot.wait_until_guild_available()
+        pydis: discord.Guild = self.bot.get_guild(Guild.id)
 
-        self.days_since_cycle = itertools.cycle([None])
+        timeout = 10  # Seconds
+        try:
+            with async_timeout.timeout(timeout):
+                await pydis.edit(**{asset_type.value: file})
```

Comment:
You are misunderstanding what the timeout does. It doesn't change assets in intervals of 10 seconds; it will raise a timeout exception and kill the upload after 10 seconds. You can see that there's an except clause below catching specifically this exception.

If you're wondering why it does that, I don't have a very good answer. [This commit](https://github.com/python-discord/sir-lancebot/commit/ab78216f4bc6cda63ae22a2538e408394c21fe30) by @scragly back in the Seasonalbot repo in 2019 added the upload utility that I've been using for inspiration since. It uses a more strict timeout for asset uploads, but doesn't explain why. I think @scragly either knew that the upload could hang for some reason, and wanted to prevent it, or just saw it as a possible weakness and wanted more control over the timeout ~ as far as I can tell, d.py uses aiohttp's default of 5 minutes.

Do you think something like this would help?
```python
with async_timeout.timeout(timeout):  # Raise after `timeout` seconds.
```

Or do you think that the timeout shouldn't be there at all?

Reply:
Yes, that comment would be really helpful, so add it.

Yeah, I can’t say if the timeout should be there or not, since am still not aware of the reason scraggly had done it, we could probably keep it.

Reply:
Comment added in 220590c.